src/librawspeed/decoders/IiqDecoder.cpp

```diff
@@ -151,13 +154,22 @@ RawImage IiqDecoder::decodeRawInternal() {
     case 0x10f:
       raw_data = bs.getSubView(data, len);
       break;
+    case 0x110:
+      correction_meta_data = bs.getSubStream(data);
```

Comment:
i *strongly* suspect
```diff
- correction_meta_data = bs.getSubStream(data);
+ correction_meta_data = bs.getSubStream(data, len);
```

Reply:
I got errors when I tried that, but that might have been another issue so I'll take another look at it.  I know the length of the new metadata section is encoded in the section itself though

Reply:
I guess https://github.com/darktable-org/rawspeed/pull/117#discussion_r153820014 will resolve those issues.